app/src/main/java/com/ivyclub/contact/ui/main/add_friend/AddFriendFragment.kt

```diff
@@ -1,8 +1,34 @@
 package com.ivyclub.contact.ui.main.add_friend
 
+import android.os.Bundle
+import android.view.View
+import android.widget.ArrayAdapter
+import androidx.fragment.app.viewModels
 import com.ivyclub.contact.R
 import com.ivyclub.contact.databinding.FragmentAddFriendBinding
 import com.ivyclub.contact.util.BaseFragment
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class AddFriendFragment : BaseFragment<FragmentAddFriendBinding>(R.layout.fragment_add_friend) {
+
+    private val viewModel: AddFriendViewModel by viewModels()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeGroups()
+    }
+
+    private fun observeGroups() {
+        viewModel.groups.observe(viewLifecycleOwner) {
+            initSpinnerAdapter(it)
+        }
+    }
+
+    private fun initSpinnerAdapter(groups: List<String>) {
+        val spinnerAdapter =
+            ArrayAdapter(requireContext(), android.R.layout.simple_spinner_item, groups)
```

Comment:
그때 멘토님이 말씀해주쎳던 requireContext() 안쓰는게 좋을거같아요!

Reply:
requireContext를 한번 없애보았는데요 혹시 이런 방법도 괜찮을까요?!

```suggestion
            context?.let { ArrayAdapter(it, android.R.layout.simple_spinner_item, groups) }
            spinnerAdapter?.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
            binding.spnGroup.adapter = spinnerAdapter
```

Reply:
근데 context가 null이면 spinnerAdapter가 초기화 안되서 문제가 없을까요? 저도 잘 모르겠네요

Reply:
spinnerAdapter가 초기화가 안되면 문제가 생기긴 할 것 같아요..